reaper.go

```diff
@@ -77,11 +77,13 @@ func NewReaper(ctx context.Context, sessionID string, provider ReaperProvider, r
 
 	c, err := provider.RunContainer(ctx, req)
 	if err != nil {
+		reaper = nil
```

Comment:
Hey @kernle32dll, thanks for taking the effort of reporting this!

I see this PR would work although I'm concerned about its maintainability if new code conditions are added in the future.

How would you see delaying the creation of the Reaper struct instance (L51) until the container and the endpoint are created (right before the old L87)? I'd see the code more maintainable in this case. wdyt?